### PR TITLE
feat: [PIE-4226]: matrix changes for pipeline studio

### PR DESCRIPTION
**Pull request by vkbansal**
##### Screenshots:

Screenshot 2022-06-22 at 6 06 54 PM
Screenshot 2022-06-22 at 6 07 44 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vkbansal** [comment]
retrigger titlecheck

**Review by kapilkumawat86 — APPROVED** [review]

**Review by devinder-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 1]
```diff
@@ -377,7 +374,7 @@ export default function FormikInputSetForm(props: FormikInputSetFormProps): Reac
                                 />
                               </GitSyncStoreProvider>
                             )}
-                            {isPipelineRemote && (
+                            {isGitSimplificationEnabled && storeType === StoreType.REMOTE && (
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:377]: Some of the changes which @depanshu-dhiman made are getting reverted here. Is it intended ?

File: src/modules/75-ci/components/PipelineStudio/BuildAdvancedSpecifications/BuildAdvancedSpecifications.tsx [hunk 2]
```diff
@@ -75,6 +78,35 @@ const BuildAdvancedSpecifications: React.FC<AdvancedSpecifications> = ({ childre
             </Layout.Horizontal>
           </Card>
         )}
+        {PIPELINE_MATRIX ? (
+          <React.Fragment>
```
Comment (chetanvekariya-harness) [inline src/modules/75-ci/components/PipelineStudio/BuildAdvancedSpecifications/BuildAdvancedSpecifications.tsx:82]: Looks like this code is exactly same as other places. If possible, later we can reuse after keeping at some common place.

**Review by chetanvekariya-harness — DISMISSED** [review]

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by chris-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/CommonPipelineStages/ApprovalStage/ApprovalStageAdvanced.tsx [hunk 3]
```diff
@@ -79,6 +84,36 @@ function ApprovalAdvancedSpecifications({
             </Layout.Horizontal>
           </Card>
         )}
+        {PIPELINE_MATRIX ? (
+          <React.Fragment>
```
Comment (mtran7) [inline src/modules/70-pipeline/components/CommonPipelineStages/ApprovalStage/ApprovalStageAdvanced.tsx:88]: shorthand can be <>

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageAdvancedInputSetForm.tsx [hunk 4]
```diff
@@ -81,6 +85,46 @@ function ConditionalExecutionFormInternal(props: ConditionalExecutionFormProps):
...
+  function handleChange(newValue: string): void {
+    try {
+      const parsed = yamlParse(newValue)
+      formik.setValues(
+        produce(formik.values, (draft: any) => {
+          set(draft, path, parsed)
+        })
+      )
+    } catch (e) {
+      // empty block
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageAdvancedInputSetForm.tsx:109]: do we need a toast when we cannot parse yaml?

File: src/modules/70-pipeline/components/PipelineStudio/LoopingStrategy/LoopingStrategy.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    open: openToggleTypeConfirmation,
+    close: closeToggleTypeConfirmation
+  } = useToggleOpen()
+
+  const initialSelectedStrategy = Object.keys(defaultTo(strategy, {}))[0] as LoopingStrategyEnum
+
+  const onTextChange = (_formikProps: FormikProps<StrategyConfig>): void => {
+    try {
+      const newValues: StrategyConfig = parse(defaultTo(yamlHandler?.getLatestYaml(), ''))
+      // formikProps.setValues(newValues)
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineStudio/LoopingStrategy/LoopingStrategy.tsx:73]: needed? or remove

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageAdvancedInputSetForm.tsx (hunk 4 above)
Comment (vkbansal) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageAdvancedInputSetForm.tsx:109]: That would be a very frequent and annoying for end user

File: src/modules/70-pipeline/components/PipelineStudio/LoopingStrategy/LoopingStrategy.tsx (hunk 5 above)
Comment (vkbansal) [reply src/modules/70-pipeline/components/PipelineStudio/LoopingStrategy/LoopingStrategy.tsx:73]: Will remove in follow up PR

File: src/modules/70-pipeline/components/CommonPipelineStages/ApprovalStage/ApprovalStageAdvanced.tsx (hunk 3 above)
Comment (vkbansal) [reply src/modules/70-pipeline/components/CommonPipelineStages/ApprovalStage/ApprovalStageAdvanced.tsx:88]: I prefer the verbose version.

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 6]
```diff
@@ -374,6 +374,21 @@ failureStrategies:
     gitBranchRequired: Git Branch is required
     gitTagRequired: Git Tag is required
     gitPRRequired: Git Pull Request is required
+loopingStrategy:
+  title: Looping Strategy
+  subTitle: 'Select 1 of the {{maxCount}} looping execution strategy below.'
+  warningText: All the configuration will be lost
```
Comment (mtran7) [inline src/modules/70-pipeline/strings/strings.en.yaml:380]: All of the configuration will be lost

**Review by mtran7 — APPROVED** [review]

**Pull request merged**